src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1876,6 +1897,74 @@ public function is_tag_closer() {
 		return $this->is_closing_tag;
 	}
 
+	/**
+	 * Returns the chunk of text from the end of the preceding tag or token to the
+	 * start of the matched tag or token, with decoded character references.
+	 *
+	 * Example:
+	 *
+	 *     $q = array( 'tag_closers' => 'visit' );
+	 *     $processor = new WP_HTML_Tag_Processor( 'Before<div>Inside</div>After' );
+	 *     $processor->next_tag( $q ); 'Before' === $processor->get_prev_text_chunk();
+	 *     $processor->next_tag( $q ); 'Inside' === $processor->get_prev_text_chunk();
+	 *     $processor->next_tag( $q ); 'After'  === $processor->get_prev_text_chunk();
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return string|null Chunk of text from end of last token to current token, or NULL if not yet matched.
+	 */
+	public function get_previous_text_chunk() {
+		if ( $this->bytes_already_parsed >= strlen( $this->html ) ) {
+			$chunk = substr( $this->html, $this->last_position->end === 0 ? 0 : $this->last_position->end + 1 );
+			$chunk = preg_replace( '/<[^a-z].*>/i', '', $chunk );
```

Comment:
What is the replacement doing here?

Reply:
one thing this raises is the need to stop at all tokens. currently, inside `next_tag()` we skip over all non-tag content, including comments and DOCTYPE and CDATA sections. the result is that we don't currently have a way to remove comments from the plain text. for now this was a cheap trick to eliminate comment matter, because at the point we have the text chunk, we have no HTML tags, meaning that almost anything that's still surrounded by `<` and `>` is a form of a comment.

this is not meant to stay.

I added code to visit comments, similar to what I did in dmsnell/wordpress-develop#7, but I think it was revealing that we need to reorient some of the internal code to think more about tokens than tags, something I believe we can do just fine without changing any external interface.

I'll remove these so that the failing tests reappear.